### PR TITLE
A quick patch for storage completion errors

### PR DESCRIPTION
**Pull request by oakeyc**
a patch for #3055 so that it doesn't dump out.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -125,15 +125,25 @@ def reg_extra_arg(self, *args, **kwargs):
...
-    account_key = parsed_args.account_key or az_config.get('storage', 'key', None)
-    connection_string = parsed_args.connection_string or az_config.get('storage', 'connection_string', None)
-    sas_token = parsed_args.sas_token or az_config.get('storage', 'sas_token', None)
-    return get_storage_data_service_client(service,
-                                           account_name,
-                                           account_key,
-                                           connection_string,
-                                           sas_token)
+    try:
+        account_name = parsed_args.account_name
```
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: `account_name = getattr(parsed_args, 'account_name', None) or az_config.get('storage', 'account', None)`
One liner without `try-except`

Comment (oakeyc) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: updated

File: src/command_modules/azure-cli-shell/azclishell/argfinder.py [hunk 2]
```diff
@@ -28,13 +28,13 @@ def get_parsed_args(self, comp_words):
...
             stderr = sys.stderr
-            sys.stderr = io.open(os.devnull, "w")
+            sys.stderr = os.open(os.devnull, "w")
 
             active_parsers[0].parse_known_args(comp_words, namespace=parsed_args)
 
             sys.stderr.close()
             sys.stderr = stderr
-        except BaseException:
+        except Exception:  # pylint: disable=broad-except
```
Comment (tjprescott) [inline src/command_modules/azure-cli-shell/azclishell/argfinder.py]: Can you use a more targeted exception here?

File: src/command_modules/azure-cli-shell/azclishell/az_completer.py [hunk 3]
```diff
@@ -222,6 +222,10 @@ def gen_dynamic_completions(self, text):
 
         except CLIError:  # if the user isn't logged in
             pass
+        except AttributeError:  # service client throws
```
Comment (tjprescott) [inline src/command_modules/azure-cli-shell/azclishell/az_completer.py]: You should be able to use `except (AttributeError, ValueError):`

File: src/command_modules/azure-cli-shell/azclishell/az_completer.py [hunk 4]
```diff
@@ -222,6 +222,8 @@ def gen_dynamic_completions(self, text):
 
         except CLIError:  # if the user isn't logged in
             pass
+        except (AttributeError, ValueError):  # service client throws
```
Comment (troydai) [inline src/command_modules/azure-cli-shell/azclishell/az_completer.py]: Why not aggregate with the `CLIError` above as well?

**Pull request merged**